clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -284,6 +294,12 @@ class MarkDeviceFunction : public RecursiveASTVisitor<MarkDeviceFunction> {
     return true;
   }
 
+  // By-pass static assert internals as everything inside it won't be found in
```

Comment:
I'd rather not do this, because this doesn't cover everything you need.  You really want to check 'isEvaluatedContext'

Reply:
MarkDeviceFunction's traversing of translation unit is only initiated in Sema::ActOnEndOfTranslationUnit. Thus, Sema::isUnevaluatedContext() won't return relevant value. Tho only solution I see here is to store values of Sema::isUnevaluatedContext() for different CallExpr-s in a map within Sema object and refer to it from markDeviceFunction::VisitCallExpr(). 

@erichkeane Maybe you know of any other viable and more preferable approach?